analysis/prep_sg.R

```diff
@@ -0,0 +1,99 @@
+####
+## This script does the following:
+# 1. Import the processed dataset of all eligible participants
+# 2. Create subsets to run subgroup analyses using the reusable action cox-ipw
+# 3. Save all subsets
+####
+
+
+# Import libraries and user functions -------------------------------------
+print('Import libraries and functions')
+library('tidyverse')
+library('here')
+library('arrow')
+
+
+# Create directories for output -------------------------------------------
+print('Create directories for output')
+fs::dir_create(here::here("output", "data", "sg"))
+
+
+# Import the data ---------------------------------------------------------
+print('Import the data')
+df <- read_feather(here("output", "data", "data_processed.arrow"))
+
+
+# Import dates ------------------------------------------------------------
+print('Import dates')
+source(here::here("analysis", "metadates.R"))
+study_dates <- lapply(study_dates, function(x) as.Date(x))
+studyend_date <- as.Date(study_dates$studyend_date, format = "%Y-%m-%d")
+
+
+# Subset the data ---------------------------------------------------------
+print('Subset the data')
+df_below60 <- df %>%
+  filter(cov_num_age < 60)
+df_60orabove <- df %>%
+  filter(cov_num_age >= 60)
+df_female <- df %>%
+  filter(cov_cat_sex == "Female")
+df_male <- df %>%
+  filter(cov_cat_sex == "Male")
+df_white <- df %>%
+  filter(cov_cat_ethnicity == "White")
+df_nonwhite <- df %>%
+  filter(cov_cat_ethnicity != "White") # includes unknown
+df_imd1 <- df %>%
```

Comment:
I wonder if the IMD subgroups would be better powered if implemented as 'IMD<5' versus 'IMD>=5' (assuming you have extracted deciles).